src/store/cache/collections/utils/addTracksFromCollections.ts

```diff
@@ -0,0 +1,58 @@
+import { makeUid } from 'utils/uid'
+import { UserCollection } from 'models/Collection'
+import { Kind } from 'store/types'
+import { reformat as reformatTrack } from 'store/cache/tracks/utils'
+import { reformat as reformatUser } from 'store/cache/users/utils'
+import { put } from 'redux-saga/effects'
+import * as cacheActions from 'store/cache/actions'
+import { ID, UID } from 'models/common/Identifiers'
+import Track from 'models/Track'
+import User from 'models/User'
+
+export function* addTracksFromCollections(
+  metadataArray: Array<UserCollection>
+) {
+  const tracks: Array<{ id: ID; uid: UID; metadata: Track }> = []
+  const users: Array<{ id: ID; uid: UID; metadata: User }> = []
+
+  metadataArray.forEach(m => {
+    if (m.tracks) {
+      m.tracks.forEach(t => {
+        tracks.push({
```

Comment:
Isn't this the same as `processAndCachTracks` which puts the action to add tracks and caches their users as well?

Reply:
yes thank you :)